Assets/Scripts/Enemy/Enemy.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Shared;
+
+class Enemy : MonoBehaviour
+{
+    [Header("Stats")]
+    [SerializeField] private float _damage;
+    [SerializeField] private float _speed;
+    [SerializeField] private float _attackSpeed;
+    private HealthSystem _health;
+
+    private bool _isAlive;
+    private const float DeathTime = 5;
+    [SerializeField] private GameObject _target;
+
+    private EnemyAnimationHandler _animationHandler;
+    private StateMachine _stateMachine;
+    private State _currentState;
+    private Dictionary<Type, Action> _stateActions;
+
+    public float Health
+    {
+        get { return _health.currentHealth; }
+        set { _health.currentHealth = value; }
+    }
+
+    public float Damage 
+    {
+        get { return _damage; }
+    }
+
+    public float Speed
+    {
+        get { return _speed; }
+    }
+
+    public bool IsAlive
+    {
+        get { return _isAlive; }
+    }
+

```

Comment:
Just make these variables public or use properties. Please no giant sections of setters and getters.

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties